src/Language/PureScript/TypeChecker/Types.hs

```diff
@@ -111,7 +111,7 @@ typesOf bindingGroupType moduleName vals = withFreshSubstitution $ do
           let solved = foldMap (S.fromList . fdDetermined) typeClassDependencies
           let constraintTypeVars = nub . foldMap (unknownsInType . fst) . filter ((`notElem` solved) . snd) $ zip (constraintArgs con) [0..]
           when (any (`notElem` unsolvedTypeVars) constraintTypeVars) $ do
-            throwError . onErrorMessages (replaceTypes currentSubst) . errorMessage $ NoInstanceFound con
+            throwError . onErrorMessages (replaceTypes currentSubst) . errorMessage $ AmbiguousTypeVariables ident generalized con
```

Comment:
Whoops; this code doesn't confirm that there are multiple instances of the constraint typeclass in existence. It seems like the strictly most correct response would be:

No instances -> NoInstanceFound
One instance -> Try to solve the type signature with that instance's type argument
Two instances -> AmbiguousTypeVariables

I'm not sure I have the expertise for that just yet, though.